Fix mobile tap behavior on Unsplash images

niranjan-uma-shankar opened the pull request:
### Summary
This PR disables the zoom-on-tap behavior for Unsplash images in mobile viewports. Instead of zooming (which doesn't work well on mobile viewports), tapping now reveals the action buttons (e.g. “Insert” and “❤️”), making the interaction more intuitive for touch users.

### Why this change?

* Zooming into an image in small viewports results in awkward aspect ratios.
* Tapping should surface actionable UI (buttons), not an unhelpful zoom state.
* On touch devices, hover-only buttons remain interactive even when invisible, resulting in confusing behaviour. For example, tapping the bottom-right corner of an image can trigger the “Insert” action, even though the button isn’t visible when the image first loads.


 This fixes https://github.com/TryGhost/Ghost/issues/24585.

### Behaviour before

Tapping an image:

* Zooms in without improving clarity
* Randomly triggers invisible hover-only buttons
* In tablet screens, no hover actions visible on zoom

| Mobile | Tablet
| ---- | ----
| <video src="https://github.com/user-attachments/assets/7c090fc3-8c91-4c3e-8d6f-dbb84652e998" /> | <video src="https://github.com/user-attachments/assets/a61446b5-0e95-453b-b344-2115a292bee1" />



### Behaviour after

This PR handles touch devices and non-touch browser resized windows correctly. 

#### In touch-enabled devices, tapping an image:

* No longer zooms in
* Consistently reveals visible action buttons
* Enables predictable interactions

| Mobile | Tablet
| ---- | ----
| <video src="https://github.com/user-attachments/assets/4742f18e-7d93-4c87-9262-47d4524accc5" /> | <video src="https://github.com/user-attachments/assets/ad34c96f-9269-4285-af49-2dd5b388f517" /> 

#### In non-touch devices, for example a desktop browser resized to a small viewport:

* The tap-to-zoom behaviour is retained up to a max-width of 540px.
* If the browser window is resized to a width below 540px, then the tap-to-zoom is disabled. Mouse hover over the images will reveal the action buttons.


https://github.com/user-attachments/assets/f4abaa13-dd67-4094-a04d-869052eb3136




### Testing instructions

1. Open ghost admin on a mobile viewport.
2. In `/posts`, create a new post and click on "Add a feature image".
3. Use Chrome devtools to simulate a mobile. Click on an image in the search result, and verify that it doesn't open the zoomed in content.
4. Verify that clicking on an image reveals the action buttons, and that each of the action buttons work as expected.
5. Search for images, and verify that the action buttons work on the search results.
6. Now use your desktop browser (not mobile simulated) and resize your browser window to smaller viewports. As you resize, verify that the mouse hover action reveals the action buttons, and clicking on an image zooms in, up to a max-width of 540px. Below 540px width, mouse hover will reveal the action buttons but the images will no longer zoom on click.

niranjan-uma-shankar committed:
Prevents zoom on touch devices

Zooming on Unsplash images in mobile and tablet devices leads to an awkward UI, not compatible with small screens. This removes the zoom capability on mobile and tablet devices.

niranjan-uma-shankar committed:
Allows clicks on overlay buttons

Overlay button clicks are allowd to pass through on mobile devices

niranjan-uma-shankar committed:
Refactors for improved maintainability

ErisDS commented:
Hi @niranjan-uma-shankar thanks for all your bug reports and fixes 🙏 
Our team are currently in code freeze whilst we're shipping 6.0. I'd really appreciate if you'd hold off raising any more until we can clear the current backlog, to avoid overwhelming the team. Thanks!

niranjan-uma-shankar committed:
Prevents small viewports in non-touch devices from zooming in

Ref https://github.com/TryGhost/Ghost/issues/24585
When the browser window on a desktop is resized to a small viewport, this fix ensures that the zoom behaviour is disabled. Zoom continues to work in tab viewports in non-touch devices.

niranjan-uma-shankar commented:
> We have two different Unsplash selector components currently, with a different [React based component](https://github.com/TryGhost/Koenig/tree/main/packages/kg-unsplash-selector) used for images inside the editor and the publication cover image in settings area, do you see a similar problem with mobile/tablet usage there?

@kevinansfield Thanks for bringing this up. I reviewed the behaviour of the React-based Unsplash modal, which is used in the lexical editor and in "Publication cover" modal in `/settings/design/edit`, and came across a few bugs:

* **UI and responsiveness bugs**: Documented in detail and addressed in [PR #1576](https://github.com/TryGhost/Koenig/pull/1576).
* **Search reliability**: There’s a race condition causing the search to return incorrect results that don’t match the search term. While the outcome is kind of similar to the Ember version bug (fixed in [Ghost PR #24583](https://github.com/TryGhost/Ghost/pull/24583)), the root cause is different. Fix in [PR #1574](https://github.com/TryGhost/Koenig/pull/1574).

#### Note

My recent PRs are functionally complete but currently in draft mode, per [this request](https://github.com/TryGhost/Ghost/pull/24586#issuecomment-3150378658) to avoid overwhelming the review queue. I’m happy to mark them “Ready for review” as soon as I [get the go-ahead](https://github.com/TryGhost/Ghost/pull/24586#issuecomment-3164399233).

kevinansfield commented:
@niranjan-uma-shankar feel free to move those PRs into ready-to-review, we're out of the 6.0 code freeze period now 🙂

niranjan-uma-shankar commented:
Hi @kevinansfield 👋 just circling back on this one. It would be great to get some eyes on this PR, along with the other related Unsplash search ones:

* https://github.com/TryGhost/Ghost/pull/24583
* https://github.com/TryGhost/Koenig/pull/1574
* https://github.com/TryGhost/Koenig/pull/1576

Happy to make any adjustments if needed, just let me know!

9larsons committed:
Aligned no-zoom cursor media query with coarse pointer detection

9larsons commented:
Hey @niranjan-uma-shankar - This looks good for the feature image selector! I see some issues with the mobile view and the Unsplash card behavior that should be handled separately. 

I see some edges where the click selection stays from image to image - I don't treat this as problematic. It's still a big move forward.
